summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/VersionedBatchStore.scala

```diff
@@ -200,13 +201,19 @@ class VersionedBatchStore[K, V, K2, V2](rootPath: String, versionsToKeep: Int, o
       case _ => Some(batchVersion)
     }
 
-    lastVals.map(pack(batchID, _))
-      .toPipe((0, 1))
-      .write(VersionedKeyValSource[K2, V2](rootPath,
-        sourceVersion = None,
-        sinkVersion = newVersion,
-        maxFailures = 0,
-        versionsToKeep = versionsToKeep))
+    val target = VersionedKeyValSource[K2, V2](rootPath,
+      sourceVersion = None,
+      sinkVersion = newVersion,
+      maxFailures = 0,
+      versionsToKeep = versionsToKeep)
+    if (!target.sinkExists(mode)) {
+      logger.info(s"Versioned batched store version for $this @ $newVersion doesn't exist. Will write out.")
+      lastVals.map(pack(batchID, _))
+        .write(target)
+    } else {
+      logger.warn(s"""Versioned batched store version for $this @ $newVersion already exists! Will skip adding to plan.
+        | Note you must have at least one store that requires writing out or the job will fail.""".stripMargin('|'))
```

Comment:
That period should be small? And your WS needs to be unreliable for this to be a thing right? It should be the small thing at the end. The only other alternative I think is to probably delete the stores when we start -- but they could be already consumed by downstream jobs on a cluster.

My understanding is the big issue is that you have a big chain of MR jobs, we finish some stores earlier than others, so if any MR job fails for any reason your sort of broken.